### PR TITLE
nixos/peertube: update configuration

### PR DESCRIPTION
**Pull request by Izorkin**
###### Description of changes
Changes in this PR:
 - Add HSTS and QUIC headers in gnix configuration.
 - Fis strartung service.

cc @marsam @mohe2015 @immae @SuperSandro2000

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/web-apps/peertube.nix [hunk 1]
```diff
@@ -68,6 +68,13 @@ let
   '';
 
   nginxCommonHeaders = ''
+    ${lib.optionalString cfg.enableWebHttps ''
+      add_header Strict-Transport-Security    'max-age=63072000; includeSubDomains';
+    ''}
+    ${lib.optionalString config.services.nginx.virtualHosts.${cfg.localDomain}.http3 ''
+      add_header Alt-Svc                      'h3=":443"; ma=86400';
+    ''}
+
```
Comment (SuperSandro2000) [inline nixos/modules/services/web-apps/peertube.nix:77]:
```suggestion
  nginxCommonHeaders = lib.optionalString cfg.enableWebHttps ''
    add_header Strict-Transport-Security    'max-age=63072000; includeSubDomains';
  '' + lib.optionalString config.services.nginx.virtualHosts.${cfg.localDomain}.http3 ''
    add_header Alt-Svc                      'h3=":443"; ma=86400';
  '' + ''
```

File: nixos/modules/services/web-apps/peertube.nix [hunk 2]
```diff
@@ -370,7 +377,7 @@ in {
     systemd.services.peertube-init-db = lib.mkIf cfg.database.createLocally {
       description = "Initialization database for PeerTube daemon";
       after = [ "network.target" "postgresql.service" ];
-      wantedBy = [ "multi-user.target" ];
+      requires = [ "postgresql.service" ];
```
Comment (SuperSandro2000) [inline nixos/modules/services/web-apps/peertube.nix:380]: You double checked that this is still started without wantedBy?

File: nixos/modules/services/web-apps/peertube.nix [hunk 3]
```diff
@@ -487,6 +496,13 @@ in {
...
             client_max_body_size                        12G;
             add_header X-File-Maximum-Size              8G always;
+
+            ${lib.optionalString cfg.enableWebHttps ''
+              add_header Strict-Transport-Security      'max-age=63072000; includeSubDomains';
+            ''}
+            ${lib.optionalString config.services.nginx.virtualHosts.${cfg.localDomain}.http3 ''
+              add_header Alt-Svc                        'h3=":443"; ma=86400';
+            ''}
           '';
```
Comment (SuperSandro2000) [inline nixos/modules/services/web-apps/peertube.nix:506]:
```suggestion
          '' + lib.optionalString cfg.enableWebHttps ''
            add_header Strict-Transport-Security      'max-age=63072000; includeSubDomains';
          '' + lib.optionalString config.services.nginx.virtualHosts.${cfg.localDomain}.http3 ''
            add_header Alt-Svc                        'h3=":443"; ma=86400';
          '';
```

File: nixos/modules/services/web-apps/peertube.nix [hunk 4]
```diff
@@ -497,6 +513,13 @@ in {
...
             client_max_body_size                        6M;
             add_header X-File-Maximum-Size              4M always;
+
+            ${lib.optionalString cfg.enableWebHttps ''
+              add_header Strict-Transport-Security      'max-age=63072000; includeSubDomains';
+            ''}
+            ${lib.optionalString config.services.nginx.virtualHosts.${cfg.localDomain}.http3 ''
+              add_header Alt-Svc                        'h3=":443"; ma=86400';
+            ''}
           '';
```
Comment (SuperSandro2000) [inline nixos/modules/services/web-apps/peertube.nix:523]:
```suggestion
          '' + lib.optionalString cfg.enableWebHttps ''
            add_header Strict-Transport-Security      'max-age=63072000; includeSubDomains';
          '' + lib.optionalString config.services.nginx.virtualHosts.${cfg.localDomain}.http3 ''
            add_header Alt-Svc                        'h3=":443"; ma=86400';
          '';
```

File: nixos/modules/services/web-apps/peertube.nix [hunk 5]
```diff
@@ -560,6 +583,13 @@ in {
...
           extraConfig = ''
             add_header Cache-Control                    'public, max-age=604800, immutable';
+
+            ${lib.optionalString cfg.enableWebHttps ''
+              add_header Strict-Transport-Security      'max-age=63072000; includeSubDomains';
+            ''}
+            ${lib.optionalString config.services.nginx.virtualHosts.${cfg.localDomain}.http3 ''
+              add_header Alt-Svc                        'h3=":443"; ma=86400';
+            ''}
           '';
```
Comment (SuperSandro2000) [inline nixos/modules/services/web-apps/peertube.nix:593]:
```suggestion
          '' + lib.optionalString cfg.enableWebHttps ''
            add_header Strict-Transport-Security      'max-age=63072000; includeSubDomains';
          '' + lib.optionalString config.services.nginx.virtualHosts.${cfg.localDomain}.http3 ''
            add_header Alt-Svc                        'h3=":443"; ma=86400';
          '';
```

File: nixos/modules/services/web-apps/peertube.nix (hunk 1 above)
Comment (Izorkin) [reply nixos/modules/services/web-apps/peertube.nix]: Fixed!

File: nixos/modules/services/web-apps/peertube.nix (hunk 2 above)
Comment (Izorkin) [reply nixos/modules/services/web-apps/peertube.nix:377]: Checking with nixos test and local virtual machine.I checked work using nixos test and on a local virtual machine.

**Pull request merged**